bats/fb-proxy-dns.bats

```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/env bats
+# vim: ft=sh:sw=2:ts=2:et
+
+set -o pipefail
+
+load os_helper
+load foreman_helper
+
+setup() {
+  tSetOSVersion
+
+	tCommandExists curl || tPackageInstall curl
+	tCommandExists dig || tPackageInstall bind-utils || tPackageInstall dnsutils
+
+  export HOSTNAME=$(hostname -f)
+}
+
+tSpRequest() {
+  if [ -d /etc/puppetlabs/puppet/ssl ] ; then
+    SP_KEY=/etc/puppetlabs/puppet/ssl/private_keys/${HOSTNAME}.pem
+    SP_CERT=/etc/puppetlabs/puppet/ssl/certs/${HOSTNAME}.pem
+    SP_CACERT=/etc/puppetlabs/puppet/ssl/ca/ca_crt.pem
+  else
+    SP_KEY=/var/lib/puppet/ssl/private_keys/${HOSTNAME}.pem
+    SP_CERT=/var/lib/puppet/ssl/certs/${HOSTNAME}.pem
+    SP_CACERT=/var/lib/puppet/ssl/ca/ca_crt.pem
+  fi
+
+  local url=https://${HOSTNAME}:8443/$1
+  shift
+  curl --silent --fail --key $SP_KEY --cert $SP_CERT --cacert $SP_CACERT $* $url
+}
+
+tGetRecord() {
+  dig +short @localhost $*
+}
+
+tVerifyRecord() {
+  local name=$1
+  local type=$2
+  local expected=$3
+
+  tFlushCache $name
+  run tGetRecord $type $name
+  [[ $status == 0 ]]
+  [[ $output == $expected ]]
+}
+
+tCreateRecord() {
+  local fqdn=$1
+  local value=$2
+  local type=$3
+  tSpRequest dns/ -X POST -d fqdn=$fqdn -d value=$value -d type=$type
+}
+
+tDeleteRecord() {
+  local name=$1
+  local type=$2
+  tSpRequest dns/$name/$type -X DELETE
+}
+
+tFlushCache() {
+  run tGetRecord ch txt version.bind
+  case $output in
+    *PowerDNS*)
+      pdns_control purge $1
+      ;;
+  esac
+}
+
+@test "verify dns feature is enabled" {
+  tSpRequest features | jq -e 'map(in({"dns": 42})) | any'
+}
+
+@test "clean up any existing records" {
+  for type in A AAAA PTR CNAME ; do
```

Comment:
```PTR``` is not needed since that's an invalid PTR here. We should drop the v4 and v6 PTRs though.